src/inventory_ui.cpp

```diff
@@ -3164,7 +3165,11 @@ void inventory_multiselector::deselect_contained_items()
     for( inventory_column *col : get_all_columns() ) {
         for( inventory_entry *selected : col->get_entries(
         []( const inventory_entry & entry ) {
-        return entry.is_item() && entry.chosen_count > 0 && entry.locations.front()->is_frozen_liquid();
+        return entry.is_item() && entry.chosen_count > 0 && ( entry.locations.front()->is_frozen_liquid() &&
+                    entry.locations.front().where() != item_location::type::map &&
+                    ( !entry.locations.front().has_parent() || ( entry.locations.front().has_parent() &&
+                            ( entry.locations.front()->length() > entry.locations.front().parent_item()->contained_where(
+                                  *entry.locations.front() )->get_pocket_data()->max_item_length ) ) ) ) ;
```

Comment:
This became very hard to understand at glance. Maybe divide this snippet into smaller chunks? Or maybe add some comments here?

Reply:
Totally agree.
I checked it after you pointed it out to me, and I'm not sure how wrote it myself, and the content is insane complex.
I will try to find a better way.

Reply:
Removed redundant conditions and added  comments.